jenkins/keti/Jenkinsfile

```diff
@@ -1,106 +1,158 @@
-				}
-				failure {
-					echo 'Build Public profile failed'
-				}
-			always {
-					junit '**/public/**/surefire-reports*/junitreports/TEST*.xml, **/public/**/surefire-reports*/**/junitreports/TEST*.xml'
-				}
-			}
-		}
+pipeline {
+    agent any
+    options {
+        buildDiscarder(logRotator(numToKeepStr:'10'))
+        skipDefaultCheckout()
+    }
+    tools {
+        maven 'apache-maven-latest'
+        jdk 'jdk1.8.0-latest'
+    }
+    stages {
+        stage('Checkout Git repositories') {
+            failFast true
+            parallel {
+                stage('Pull request') {
+                    when {
+                        changeRequest()
+                    }
+                    steps {
+                        dir('keti') {
+                            checkout(
+                                [ $class: 'GitSCM',
+                                  branches: [ [ name: env.CHANGE_BRANCH ] ],
+                                  doGenerateSubmoduleConfigurations: false,
+                                  extensions: [ [ $class: 'SubmoduleOption',
+                                                  disableSubmodules: false,
+                                                  parentCredentials: true,
+                                                  recursiveSubmodules: true,
+                                                  reference: '',
+                                                  trackingSubmodules: true ] ],
+                                  submoduleCfg: [ ],
+                                  userRemoteConfigs: [ [ credentialsId: '670c4a5c-7db1-4a51-9706-e49458224cda',
+                                                         url: 'https://github.com/eclipse/keti' ] ] ]
+                            )
+                        }
+                    }
+                }
+                stage('Non-pull-request') {
+                    when {
+                        not {
+                            changeRequest()
+                        }
+                    }
+                    steps {
+                        dir('keti') {
+                            checkout(
+                                [ $class: 'GitSCM',
+                                  branches: [ [ name: env.BRANCH_NAME ] ],
+                                  doGenerateSubmoduleConfigurations: false,
+                                  extensions: [ [ $class: 'SubmoduleOption',
+                                                  disableSubmodules: false,
+                                                  parentCredentials: true,
+                                                  recursiveSubmodules: true,
+                                                  reference: '',
+                                                  trackingSubmodules: true ] ],
+                                  submoduleCfg: [ ],
+                                  userRemoteConfigs: [ [ credentialsId: '670c4a5c-7db1-4a51-9706-e49458224cda',
```

Comment:
We should not hardcode credentials here.  Add it to Jenkins credentials instead.

Reply:
This is autogenerated by Jenkin's [Snippet Generator](https://ci.eclipse.org/keti/job/Keti-Build/view/change-requests/job/PR-23/pipeline-syntax/) so in the event the backing credentials change we can always regenerate the code.